core/ecschema-editing/src/Editing/Formats.ts

```diff
@@ -28,7 +28,10 @@ export class Formats extends SchemaItems {
 
       if (units !== undefined) {
         for (const unit of units) {
-          const unitItem = await this.getSchemaItem<Unit | InvertedUnit>(unit, null);
+          const unitItem =  await this.schemaEditor.schemaContext.getSchemaItem<Unit | InvertedUnit>(unit);
+          if (!unitItem) {
+            throw new SchemaEditingError(ECEditingStatus.SchemaItemNotFoundInContext, new SchemaItemId(SchemaItemType.Unit, unit));
+          }
```

Comment:
Actually, I think we could also use the new method of SchemaContextEditor.getSchemaItem here. It uses the same method to get the item and also throws an error if the returned value is undefined with the same erro rmessage.

Reply:
The reason I can't is because it could be either unit or invertedUnit.  If you look at the PR diff, in the old code I had a hacky work around where if you passed in null it would bypass the SchemaItemType check.  I decided this was to ugly, so I just replaced it with the underlying editor call that doesn't have validation.  I also experimented with passing in an array of schemaItemTypes, but thought it was to ugly as well.  I'm only do this in two places, both involving the unit/invertedunit scenario.